docs/source/reference/ha.rst

```diff
@@ -79,18 +79,21 @@ sensors is likely to be an enhancement to |st2| in future releases.
 
 st2rulesengine
 ^^^^^^^^^^^^^^
-This is a dual purpose process. Its main function is to evaluate rules when it sees
+``st2rulesengine`` evaluates rules when it sees
 TriggerInstances and decide if an ActionExecution is to be requested. It needs access to MongoDB to
-locate rules and RabbitMQ to listen for TriggerInstances and request ActionExecutions. The
-auxiliary purpose of this process is to run all the defined timers. See 
-:ref:`timers <ref-rule-timers>` for the specifics on setting up timers via rules.
+locate rules and RabbitMQ to listen for TriggerInstances and request ActionExecutions.
 
 Multiple ``st2rulesengine`` processes can run in active-active with only connections to MongoDB and
 RabbitMQ. All these will share the TriggerInstance load and naturally pick up more work if one or
-more of the processes becomes unavailable. The timer function in each of the RulesEngine is not HA 
-compatible. In the interim it is possible to disable the timer in all but one of the
-``st2rulesengine`` to avoid duplicate timer events. Expect this to be fixed in a future |st2|
-release.
+more of the processes becomes unavailable.
+
+st2timersengine
+^^^^^^^^^^^^^^^
+
+``st2timersengine`` is responsible for scheduling all user specified timers. See
+:ref:`timers <ref-rule-timers>` for the specifics on setting up timers via rules.
+
+You have to have one active ``st2timersengine`` process running to schedule all timers. This is trivial to setup in Kubernetes so there is exactly one active container running ``st2timersengine`` process. Failover is handled natively by Kubernetes. In non Kubernetes deployments, external monitoring needs to setup and a new ``st2timersengine`` process needs to be spun up to address failover.
```

Comment:
In HA doc I don't think we should mention any Kubernetes specifics, but have general descriptions as we do for all other services.

Reply:
Yeah, I thought about it but then we need to say how to handle failover. Should we just leave out the k8s part?

Reply:
>While Kubernetes can "guarantee something", obviously container/process could be killed by whatever reasons and there is no other timersengine that will keep running.
Won't this be taken care of by Kubernetes though? Isn't that the whole point of using it? 

>This sounds like we don't have any way to promise timers in HA. I mean can't run at least 2+ instances.
Correct, this is what we decided to do with timers. If we decide to solve this, then we need to look at leader election which we intentionally decided to avoid. See https://github.com/StackStorm/discussions/issues/305

>I think it's worth mentioning what happens in scenario when timersengine is not available, if we can't guarantee HA for it. 
There is no A at that point leave alone HA :). It goes without saying IMO but I'll make that explicit. 

>Additionally, what's needed for timersengine to run properly (DB, MQ, anything else), as it's described for other services. Are there any other services that rely on timers functionality?
+1

Reply:
Pod/container failure will lead to reschedule/restart. What happens in between is a downtime.

Related to that, we'll need to add `/status` for each st2 service https://github.com/StackStorm/k8s-st2/issues/5 so K8s can control the reschedule and knows whether the service is really alive or just sits there in any manner of deadlock, spinning the cycles and being actually non-responsive. See https://github.com/StackStorm/st2/issues/4020


With no big A in HA, we still can haz High :smiley: